### PR TITLE
sys/riotboot/flashwrite.c: add message to static assert 

### PR DESCRIPTION
**Pull request by ant9000**
Fix for issue #16043

### CODE REVIEW

File: sys/riotboot/flashwrite.c [hunk 1]
```diff
@@ -47,11 +47,13 @@ int riotboot_flashwrite_init_raw(riotboot_flashwrite_t *state, int target_slot,
...
     /* the flashpage size must be a multiple of the riotboot flashpage buffer */
-    static_assert(!(FLASHPAGE_SIZE % RIOTBOOT_FLASHPAGE_BUFFER_SIZE));
+    static_assert(!(FLASHPAGE_SIZE % RIOTBOOT_FLASHPAGE_BUFFER_SIZE),
+        "Flashpage size must be a multiple of riotboot flashpage buffer.");
 #else
     /* The flashpage buffer must be a multiple of the write block size */
     static_assert(!(RIOTBOOT_FLASHPAGE_BUFFER_SIZE %
-                    FLASHPAGE_WRITE_BLOCK_SIZE));
+        FLASHPAGE_WRITE_BLOCK_SIZE),
+        "Flashpage buffer must be a multiple of write block size.");
```
Comment (fjmolinas) [inline sys/riotboot/flashwrite.c:56]: Can you address uncrustify suggestions?

**Review by fjmolinas — COMMENTED** [review]
Thanks for the fix @ant9000, but I wonder why does this only surfaces with `bootloaders/riotboot_dfu`? Is it because of compilation flags?

**Comment by ant9000** [comment]
> Thanks for the fix @ant9000, but I wonder why does this only surfaces with `bootloaders/riotboot_dfu`? Is it because of compilation flags?

Found out it's a toolchain issue - Debian Buster gcc for arm chokes on that static_assert define.

File: sys/riotboot/flashwrite.c (hunk 1 above)
Comment (ant9000) [reply sys/riotboot/flashwrite.c]: Should be fixed.

**Comment by fjmolinas** [comment]
> > Thanks for the fix @ant9000, but I wonder why does this only surfaces with `bootloaders/riotboot_dfu`? Is it because of compilation flags?
> 
> Found out it's a toolchain issue - Debian Buster gcc for arm chokes on that static_assert define.

It's indeed the toolchain, I was able to reproduce by downgrading mine. I'm not ACK'ing the fix yet because this should work as-is , I'm trying to understand why this older toolchain is complaining here.

**Comment by ant9000** [comment]
> > > Thanks for the fix @ant9000, but I wonder why does this only surfaces with `bootloaders/riotboot_dfu`? Is it because of compilation flags?
> > 
> > 
> > Found out it's a toolchain issue - Debian Buster gcc for arm chokes on that static_assert define.
> 
> It's indeed the toolchain, I was able to reproduce by downgrading mine. I'm not ACK'ing the fix yet because this should work as-is , I'm trying to understand why this older toolchain is complaining here.

According to this [reference](https://en.cppreference.com/w/cpp/language/static_assert), _static_assert_  as defined in C++11 requires both an expression and a message. Message is only optional from C++17 onwards.

A quick grep throughout RIOT sources shows it's indeed used with a message everywhere - apparently I stumbled upon the only non C+11 usage around.
https://en.cppreference.com/w/cpp/language/static_assert

**Comment by ant9000** [comment]
Same goes for ISO/IEC 9899:201x standard - message is not optional: 

http://www.open-std.org/Jtc1/sc22/wg14/www/docs/n1548.pdf

**Comment by fjmolinas** [comment]
> > > > Thanks for the fix @ant9000, but I wonder why does this only surfaces with `bootloaders/riotboot_dfu`? Is it because of compilation flags?
> > > 
> > > 
> > > Found out it's a toolchain issue - Debian Buster gcc for arm chokes on that static_assert define.
> > 
> > 
> > It's indeed the toolchain, I was able to reproduce by downgrading mine. I'm not ACK'ing the fix yet because this should work as-is , I'm trying to understand why this older toolchain is complaining here.
> 
> According to this [reference](https://en.cppreference.com/w/cpp/language/static_assert), _static_assert_ as defined in C++11 requires both an expression and a message. Message is only optional from C++17 onwards.
> 
> A quick grep throughout RIOT sources shows it's indeed used with a message everywhere - apparently I stumbled upon the only non C++11 usage around.

Ahh nice, thanks for the explanation. Only one comment then, do you mind reformating the commit message according to our coding conventions https://github.com/RIOT-OS/RIOT/blob/master/CONTRIBUTING.md#commit-conventions, something like:

```
sys/riotboot/flashwrite.c: add message to static assert

static_assert as defined in c11 requires both an expression and a message.
The message is only optional from c17 onwards.
```

**Review by fjmolinas — APPROVED** [review]
ACK, all good thanks for the fix @ant9000:

```
arm-none-eabi-gcc --version
arm-none-eabi-gcc (GNU Tools for Arm Embedded Processors 7-2017-q4-major) 7.2.1 20170904 (release) [ARM/embedded-7-branch revision 255204]
Copyright (C) 2017 Free Software Foundation, Inc.
This is free software; see the source for copying conditions.  There is NO
warranty; not even for MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
```

```
BOARD=samr34-xpro make -C bootloaders/riotboot_dfu/ all 
Building application "riotboot_dfu" for "samr34-xpro" with MCU "saml21".

"make" -C /home/francisco/workspace/RIOT/boards/samr34-xpro
"make" -C /home/francisco/workspace/RIOT/core
"make" -C /home/francisco/workspace/RIOT/cpu/saml21
"make" -C /home/francisco/workspace/RIOT/cpu/cortexm_common
"make" -C /home/francisco/workspace/RIOT/cpu/cortexm_common/periph
"make" -C /home/francisco/workspace/RIOT/cpu/sam0_common
"make" -C /home/francisco/workspace/RIOT/cpu/sam0_common/periph
"make" -C /home/francisco/workspace/RIOT/cpu/saml21/periph
"make" -C /home/francisco/workspace/RIOT/drivers
"make" -C /home/francisco/workspace/RIOT/drivers/periph_common
"make" -C /home/francisco/workspace/RIOT/sys
"make" -C /home/francisco/workspace/RIOT/sys/checksum
"make" -C /home/francisco/workspace/RIOT/sys/div
"make" -C /home/francisco/workspace/RIOT/sys/event
"make" -C /home/francisco/workspace/RIOT/sys/fmt
"make" -C /home/francisco/workspace/RIOT/sys/luid
"make" -C /home/francisco/workspace/RIOT/sys/malloc_thread_safe
"make" -C /home/francisco/workspace/RIOT/sys/newlib_syscalls_default
"make" -C /home/francisco/workspace/RIOT/sys/pm_layered
"make" -C /home/francisco/workspace/RIOT/sys/riotboot
"make" -C /home/francisco/workspace/RIOT/sys/stdio_uart
"make" -C /home/francisco/workspace/RIOT/sys/usb/usbus
"make" -C /home/francisco/workspace/RIOT/sys/usb/usbus/dfu
"make" -C /home/francisco/workspace/RIOT/sys/xtimer
   text	   data	    bss	    dec	    hex	filename
  11520	     36	   3896	  15452	   3c5c	/home/francisco/workspace/RIOT/bootloaders/riotboot_dfu/bin/samr34-xpro/riotboot_dfu.elf
```

**Pull request merged**